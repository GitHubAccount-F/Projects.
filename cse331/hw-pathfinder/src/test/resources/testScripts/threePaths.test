#Tests Dijkstra's Algorithm when there is three
#possible paths to a target node

#Creates graph
CreateGraph graph1

#Adds the nodes
AddNode graph1 one
AddNode graph1 two
AddNode graph1 three
AddNode graph1 four
AddNode graph1 five
AddNode graph1 six
AddNode graph1 seven
AddNode graph1 eight

#Adds edges
AddEdge graph1 one eight 20
AddEdge graph1 one two 4
AddEdge graph1 two three 4
AddEdge graph1 three eight 4
AddEdge graph1 one four 2
AddEdge graph1 four five 2
AddEdge graph1 five six 2
AddEdge graph1 six seven 2
AddEdge graph1 seven eight 2

#Finds the least costly path
FindPath graph1 one eight