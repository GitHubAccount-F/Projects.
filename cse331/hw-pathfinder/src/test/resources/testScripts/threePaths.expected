#Tests Dijkstra's Algorithm when there is three
#possible paths to a target node

#Creates graph
created graph graph1

#Adds the nodes
added node one to graph1
added node two to graph1
added node three to graph1
added node four to graph1
added node five to graph1
added node six to graph1
added node seven to graph1
added node eight to graph1

#Adds edges
added edge 20.000 from one to eight in graph1
added edge 4.000 from one to two in graph1
added edge 4.000 from two to three in graph1
added edge 4.000 from three to eight in graph1
added edge 2.000 from one to four in graph1
added edge 2.000 from four to five in graph1
added edge 2.000 from five to six in graph1
added edge 2.000 from six to seven in graph1
added edge 2.000 from seven to eight in graph1

#Finds the least costly path
path from one to eight:
one to four with weight 2.000
four to five with weight 2.000
five to six with weight 2.000
six to seven with weight 2.000
seven to eight with weight 2.000
total cost: 10.000